### PR TITLE
Move to using primary constructors.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/MemberInfo.cs [hunk 1]
```diff
@@ -6,26 +6,20 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.UseIndexOrRangeOperator
 {
-    internal readonly struct MemberInfo
+    internal readonly struct MemberInfo(
```
Comment (huoyaoyuan) [inline src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/MemberInfo.cs:9]: Why can't we use records here? The concerns I can see are assembly size and difficulty for documentation.

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/MemberInfo.cs:9]: > Why can't we use records here?

No one said we couldn't use records here :-)

**Comment by CyrusNajmabadi** [comment]
@ToddGrun a mechanical change with teh code fix provider.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**